lib/ssl/test/ssl_mfl_SUITE.erl

```diff
@@ -0,0 +1,530 @@
+                          ConfigProtocol ->
+                              ConfigProtocol
+                      end,
+
+    ssl_test_lib:wait_for_openssl_server(Port,  OpensslProtocol),
+
+    Client0 =
+        ssl_test_lib:start_client([{node, ClientNode},
+                                   {port, Port}, {host, Hostname},
+                                   {mfa, {ssl_test_lib, session_id, []}},
+                                   {from, self()},
+                                   {options, [{reuse_sessions, save}, {verify, verify_peer}| ClientOpts]}]),
+
+    SID = receive
+              {Client0, Id0} ->
+                  Id0
+          end,
+
+    %% quit s_server's current session so we can interact with the next client
+    true = port_command(OpensslPort, "q\n"),
+    ssl_test_lib:close(Client0),
+
+    flush(),
+
+    Client1 =
+        ssl_test_lib:start_client([{node, ClientNode},
+                                   {port, Port}, {host, Hostname},
+                                   {mfa, {ssl_test_lib, session_id, []}},
+                                   {from, self()},  {options, [{reuse_session, SID} | ClientOpts]}]),
+    receive
+        {Client1, SID} ->
+            ok
+    after ?SLEEP ->
+            ct:fail(session_not_reused)
+    end,
+
+    ErlRecvFun = fun() ->
+                         Data = ssl_test_lib:check_active_receive(Client1, Data)
+                 end,
+    run_mfl_openssl(Client1, OpensslPort, MFL, Data, ErlRecvFun),
+
+    %% Clean close down!   Server needs to be closed first !!
+    ssl_test_lib:close_port(OpensslPort),
+    ssl_test_lib:close(Client1).
+
+%%--------------------------------------------------------------------
+
+handshake_continue(Config) when is_list(Config) ->
+    ok = run_mfl_handshake_continue(Config, 1024),
+    ok = run_mfl_handshake_continue(Config, undefined),
+    ok.
+
+run_mfl_handshake_continue(Config, MFL) ->
+    Data = if is_integer(MFL) ->
+                   "hello world" ++ lists:duplicate(MFL, $u);
+              true ->
+                   "hello world" ++ lists:duplicate(999, $x)
+           end,
+    ClientExtraOpts = [{handshake, hello}, {max_fragment_length, MFL}],
+    ServerExtraOpts = [{handshake, hello}],
+    ExtraStartOpts = [{continue_options, [{want_ext, self()}]}],
+    MflEnum = mfl_enum(MFL),
+    PostF = fun(Server, Client) ->
+                    receive {Server, {ext, ServerExt}} ->
+                            ct:log("Server handshake Ext ~p~n", [ServerExt]),
+                            MflEnum = maps:get(max_frag_enum, ServerExt, undefined)
+                    end,
+                    receive {Client, {ext, ClientExt}} ->
+                            ct:log("Client handshake Ext ~p~n", [ClientExt]),
+                            case maps:get(server_hello_selected_version, ClientExt, undefined) of
+                                {3,4} ->
+                                    %% For TLS 1.3 the ssl {handshake, hello} API is inconsistent:
```

Comment:
This is a todo item for us. @IngelaAndin

Reply:
Yes, the API was created before TLS-1.3 and hence TLS-1.3 differences where not considered. We are aware that considerations for TLS-1.3 are needed, but have not prioritized that yet.